src/slots/Slot.js

```diff
@@ -103,13 +115,16 @@ const SCHEMA = Joi.object({
           is: DATA_TYPES.INTEGER,
           then: Joi.number().integer().strict().allow(null),
         },
-        {
-          is: DATA_TYPES.ANY,
-          then: Joi.number().integer().strict().allow(null),
-        },
         {
           is: DATA_TYPES.FLOAT,
           then: Joi.number().strict().allow(null),
+        },
+        {
+          is: DATA_TYPES.ANY,
+          then: Joi.number().integer().strict().allow(null),
+        }, {
+          is: DATA_TYPES.ANY_NUMBER,
+          then: Joi.number().allow(null),
```

Comment:
you should make this `strict`, otherwise numbers as strings will be valid to, like `'123'`. 
This is because Joi was originally designed for things like query string parameters, where numbers and booleans are strings, too.